### PR TITLE
#ft-162414269 All validations

### PR DESCRIPTION
**Pull request by ebzeal**
#### What does this PR do?
 - Custom Validations for all endpoints
 - Sanitizes endpoints
#### Description of Task to be completed?
 - The validations are used as middleware on endpoints. Some of these endpoints include:
   post  - api/auth/signup
   post  - api/auth/login
   post  - api/articles/:slug/comments
   put    - api/articles/:slug/comments/:commentId
   put    - api/users',

#### How should this be manually tested?
 - Test each of the endpoints above

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2229851/stories/162414269

### CODE REVIEW

**Comment by Makwe-O** [comment]
LGTM, Yet

**Review by rafmme — COMMENTED** [review]
LGTM, yet

**Comment by tejiri4** [comment]
LGTM, yet.

File: server/controllers/ArticleController.js [hunk 1]
```diff
@@ -252,6 +253,22 @@ class ArticleController {
...
+   * @param {object} req HTTP request object
+   * @param {object} res HTTP response object
+   * @returns {object} api route response
+   */
+  static async search(req, res) {
+    const { author } = req.query;
+    if (author) {
+      SearchController.byAuthor(author, res);
+    } else {
+      return response(res, 400, 'failure', 'No search parameters supplied', null, null);
```
Comment (Bobsar0) [inline server/controllers/ArticleController.js]: No need for the null values at the end

File: server/controllers/UserController.js [hunk 2]
```diff
@@ -222,10 +224,14 @@ class UserController {
    */
   static async logIn(req, response) {
     try {
-      const { email, password } = req.body;
+      const { userNameEmail, password } = req.body;
 
       const user = await User.findOne({
-        where: { email }
+        // where: { $or: [{ email: userNameEmail }, { userName: userNameEmail }] }
```
Comment (Bobsar0) [inline server/controllers/UserController.js]: Delete comment

File: server/controllers/UserController.js [hunk 3]
```diff
@@ -222,10 +224,14 @@ class UserController {
...
+        where: {
+          [Op.or]: [{ email: userNameEmail }, { userName: userNameEmail }]
+          // $or: [{ a: 5 }, { a: 6 }]
```
Comment (Bobsar0) [inline server/controllers/UserController.js]: Delete comment

Comment (ebzeal) [reply server/controllers/UserController.js]: yeah, thanks

File: server/middlewares/validations/ArticleRatingValidations.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { validationResult, checkSchema } from 'express-validator/check';
+
+const handleValidationErrors = (req, res, next) => {
```
Comment (Bobsar0) [inline server/middlewares/validations/ArticleRatingValidations.js]: This is called multiple times in several files. Consider making a helper function to aid re-usability?

Comment (ebzeal) [reply server/middlewares/validations/ArticleRatingValidations.js]: sorted

File: server/controllers/ArticleController.js (hunk 1 above)
Comment (ebzeal) [reply server/controllers/ArticleController.js]: oh, not part of my PR file changes though, but I'll check

**Review by seunkoko — APPROVED** [review]

**Pull request merged**